src/lib/common/serial.c

```diff
@@ -237,19 +239,22 @@ void serialGenericUartIRQHandler(SerialHandle_t *handle) {
       usart_IsEnabledIT_TXE((USART_TypeDef *)handle->device)) {
     uint8_t txByte;
     configASSERT(handle->getTxBytesFromISR);
-    size_t bytesAvailable = handle->getTxBytesFromISR(handle, &txByte, 1);
+    bytesAvailable = handle->getTxBytesFromISR(handle, &txByte, 1);
 
     if (bytesAvailable > 0) {
       // Transmit current byte
       usart_TransmitData8((USART_TypeDef *)handle->device, txByte);
-    } else   if(LL_USART_IsActiveFlag_TC((USART_TypeDef *)handle->device)) {
+    } else {
+      usart_DisableIT_TXE((USART_TypeDef *)handle->device);
+    }
+  }
+
+  if (!bytesAvailable && LL_USART_IsActiveFlag_TC((USART_TypeDef *)handle->device) && !usart_IsEnabledIT_TXE((USART_TypeDef *)handle->device)) {
```

Comment:
This is the heart of the fix.

Instead of waiting for the TC(transmission complete) interrupt while leaving the TXE(transmission buffer empty) interrupt enabled, we disable the TXE interrupt when there are 0 bytes left to send. We will then only interrupt when we get a TC flag.

Previously, because we didn't disable the TXE interrupt, the TXE flag would cause us to continually execute the interrupt. Luckily, disabled the TXE interrupt does not "clear" the flag, therefore the TC interrupt, which relies on the TXE flag being set still triggers an interrupt.